src/rgw/CMakeLists.txt

```diff
@@ -259,6 +259,7 @@ if(WITH_RADOSGW_POSIX)
   list(APPEND librgw_common_srcs
 	      driver/posix/rgw_sal_posix.cc
 	      driver/posix/lmdb-safe.cc
+	      driver/posix/posixDB.cc
```

Comment:
I'm guessing this will fail to link if you cmake with --WITH_RADOSGW_DBSTORE=NO.  Can you test and fix that, if it's the case?